src/routes/safe/store/actions/transactionsNew/loadAllTransactions.ts

```diff
@@ -0,0 +1,90 @@
+import axios, { AxiosResponse } from 'axios'
+
+import { getNewTransactionsServiceUriFrom, getTxServiceHost } from 'src/config'
+import { checksumAddress } from 'src/utils/checksumAddress'
+import { Transaction } from '../../models/types/transactions'
+
+export type ServiceUriParams = {
+  safeAddress: string
+  limit: number
+  offset: number
+  orderBy?: string // todo: maybe this should be key of MultiSigTransaction | keyof EthereumTransaction
+  queued?: boolean
+  trusted?: boolean
+}
+
+type EndpointResponse = {
+  count: number
+  next?: string
+  previous?: string
+  results: Transaction[]
+}
+
+const getAllTransactionsUrl = (safeAddress: string) => {
+  const host = getTxServiceHost()
+  const address = checksumAddress(safeAddress)
+  const base = getNewTransactionsServiceUriFrom(address)
+
+  return `${host}${base}`
+}
+
+const fetchAllTransactions = async (
+  urlParams: ServiceUriParams,
+  eTag: string | null,
+): Promise<{ responseEtag: string; results: Transaction[]; count?: number }> => {
```

Comment:
can we move this `{ responseEtag: string; results: Transaction[]; count?: number }` to a separated type and export it? seems like something we could make use of in some other file